--- firstProject.cpp	2019-06-17 15:25:26.000000000 +0800
+++ firstProject_patch.cpp	2019-06-22 00:20:06.000000000 +0800
@@ -103,7 +103,7 @@
     }
     ~PM()
     {
-        delete[] project;
+        //delete[] project;
         salary = 0;
     }
     char *get_project()
@@ -269,7 +269,7 @@
                 cout << "Salary :";
                 cin >> salary;
                 cout << "Done !" << endl;
-                *pm_count++;
+                (*pm_count)++;
                 pm_array[idx] = new PM(name, age, project, salary);
                 tmpname = new char[strlen(name.c_str())];
                 memcpy(tmpname, name.c_str(), strlen(name.c_str()));
@@ -297,7 +297,7 @@
                 cout << "Salary :";
                 cin >> salary;
                 cout << "Done !" << endl;
-                *rd_count++;
+                (*rd_count)++;
                 rd_array[idx] = new RD(name, age, language, salary);
                 tmpname = new char[strlen(name.c_str())];
                 memcpy(tmpname, name.c_str(), strlen(name.c_str()));
@@ -325,7 +325,7 @@
                 cout << "Salary :";
                 cin >> salary;
                 cout << "Done !" << endl;
-                *hr_count++;
+                (*hr_count)++;
                 hr_array[idx] = new HR(name, age, department, salary);
                 tmpname = new char[strlen(name.c_str())];
                 memcpy(tmpname, name.c_str(), strlen(name.c_str()));
@@ -394,7 +394,7 @@
         cout << "index :";
 
         idx = read_int();
-        if (idx < MAX && pm_array[idx])
+        if (idx >=0 && idx < MAX && pm_array[idx])
         {
             PM *tpm = NULL;
             tpm = pm_array[idx];
@@ -407,7 +407,7 @@
         cout << "index :";
         idx = read_int();
 
-        if (idx < MAX && rd_array[idx])
+        if (idx >=0 && idx < MAX && rd_array[idx])
         {
             RD *trd = NULL;
             trd = rd_array[idx];
@@ -418,7 +418,7 @@
     case 3:
         cout << "index :";
         idx = read_int();
-        if (idx < MAX && hr_array[idx])
+        if (idx >=0 && idx < MAX && hr_array[idx])
         {
             HR *thr = NULL;
             thr = hr_array[idx];
